### DISCUSSION TITLE
If another window is displayed (becomes active)

### DISCUSSION THREAD

**Discussion by pcevent** [slint-ui/slint#2979] [Q&A] [answered]
I want to make another window ui1 display when I click a button on the current window ui0 (if ui1 is minimized, have it cancel the minimized state). I tried to call the ui1->show() method, but it didn't work. What can I do to achieve this function?

**Comment by ogoffart** [comment]
This is not yet implemented. 
I believe we would need to add a `raise()` function to the window, or something like that

**Comment by pcevent** [comment]
Okay, thank you.

**Comment by ogoffart** [comment, marked as answer]
Right now, it is possible with private API using the i-slint-backend-winit crate.
You need to enable it like so in your Cargo.toml file

```toml
i-slint-backend-winit = "=1.1.0"
```

with the "=" , because we don't follow semver for internal crates.
then you can do something like

```rust
fn main() {
    // make sure the winit platform is initialized (as opposed to an other platform) by setting it before any other slint call
    slint::platform::set_platform(Box::new(i_slint_backend_winit::Backend::new()));
    
    // ...

    i_slint_backend_winit::WinitWindowAccessor::with_winit_window(ui1.window(), |win| win.focus_window());
}
```

(I haven't tried this)
But this use the [winit::Window](https://docs.rs/winit/0.28.6/winit/window/struct.Window.html) API and allow to have more control over the window.

**Reply by Vadoola** [reply]
> (I haven't tried this)
But this use the [winit::Window](https://docs.rs/winit/0.28.6/winit/window/struct.Window.html) API and allow to have more control over the window.

Just as a comment about this technique, so far in my limited testing it seems to work (for my purpose). I'm working on a toy project to learn slint better, and it's using a borderless window and re-creating the close and minimize buttons. I couldn't figure out how to actually minimize the window, but using this technique I can call winit set_minimized, so far I haven't seen any issues.

**Comment by pcevent** [comment]
Okay, that's great! Thank you!

**Comment by pcevent** [comment]
Using the above method, a bug occurred on the v1.1.0 backend instead. TouchArea in Flickable has slowed down its response to clicked events. When the mouse is clicked and raised too quickly, this event is not triggered. I tried v1.0.2 again and it can trigger normally.

**Reply by ogoffart** [reply]
Could you please submit a bug report on our issue tracker? Including a reproducible test case that demonstrates the issue would be immensely helpful to investigate and address the problem.

(There was some change related to flickable and event handling so this could explain it)

**Comment by pcevent** [comment]
The TouchArea above is normal, while the TouchArea below is normal in version 1.0. Exception in version 1.1 (pressing the left mouse button slowly to release can respond to events, pressing the left mouse button immediately to release does not respond to events). The code is as follows:
```rust
export component AppWindow inherits Window {
    width: 200px;
    height: 100px;
    VerticalBox {
        Rectangle {
            height: 50px;
            Text {
                text: "ok";
            }
            TouchArea {
                clicked => {
                    debug("ok")
                }
            }
        }
        Flickable {
            Rectangle {
                height: 50px;
                Text {
                    text: "??";
                }
                TouchArea {
                    clicked => {
                        debug("??")
                    }
                }
            }
        }
    }
}
```

**Reply by pcevent** [reply]
In version v1.1. Release immediately after pressing the left button. The event was not captured.

**Reply by ogoffart** [reply]
I can't reproduce this problem. 

There was one change in 1.1 is that clicked is only going to be called if the TouchArea was also pressed. But this doesn't seem to be the problem you describe.

I do get the "??" even when clicking really quicly on it.

**Reply by pcevent** [reply]
This is the project source file:
[slint-touch.zip](https://github.com/slint-ui/slint/files/11938219/slint-touch.zip)

This is the generated executable file:
[slint-touch-bin.zip](https://github.com/slint-ui/slint/files/11938233/slint-touch-bin.zip)

Quickly click the mouse.

**Comment by Vadoola** [comment]
So I have since discovered an issue doing this that appears to be specific to the winit backend and Wayland.

I haven't dug too deep, but as mentioned above, in my project I create a borderless window and recreate the window frame to theme it with the rest of the application. I need access to the backend so I can programmatically minimize it, and hide the window to "minimize" it to the tray.

When I first added in the use of i_slint_backend_winit, I noticed no issues, however I was testing on Windows 11 at the time.
forcing the Winit backend on Linux seems to work fine on X11, but very subtlety breaks on Wayland.  Under Wayland using the winit backend I loose the ability to move my Window around by grabbing the title bar, but everything else seems to work. On X11 and Windows it continues to work just fine. If I switch the backend from winit to Qt it seems to work on Wayland as well, although now the app ignores the scaling on my HiDPI monitor.

I'm not sure entirely sure if this is relative to this discussion or I should be opening a new discussion / issue, but It came up because I was using the technique mentioned in this discussion.

I need to try and get a video of it in action, and narrow it down to a minimum example, but I thought I would comment here and see if there were any ideas while I try and work on that.

**Reply by ogoffart** [reply]
I think this is indeed an unrelated issue.

I'm not sure I understand really what the problem is
It could be a bug in winit. But i'm not sure how you do the title bar so i'm not sure what's broken.

**Reply by Vadoola** [reply]
No worries, I'm trying to quickly hack together a minimum viable example, and will open an Issue. I'm currently leaning towards it might be a Winit issue, but I know very little of the internals of winit. Perhaps you guys can take a look, and if it is with winit, maybe point me in the correct direction to try and file an issue upstream?

**Comment by pcevent** [comment]
May I ask if the software package I uploaded is also running normally on your end? This is a software package on MacOS.
